Fix issues with adding the background graphic and then merge

realsimonwu opened the pull request:
In this branch so far I've made some small changes to `TopSection` in order to accommodate the background graphic, and I've added the background graphic as a `background-image` to `PageLayout`. However, at smaller breakpoints in viewport size, a gap appears between the background graphic and the navbar.

realsimonwu committed:
Add background graphic and some small changes to `TopSection`

Added the background graphic file to `public/home`, and made some small changes to `TopSection` to accomodate for the top graphic. Note that this still has a lot of issues right now at small viewport breakpoints that I'm trying to resolve.

realsimonwu committed:
Fix the gap that appears at small breakpoints

realsimonwu committed:
Run the linter on everything

realsimonwu commented:
I also changed the background graphics to the dark navy colour as per Sharon's request: it conflicts with some of the other graphics but that can be resolved later

realsimonwu committed:
Update TopSection.jsx

realsimonwu committed:
Update TopSection.jsx

realsimonwu commented:
Oh yeah, I did fix the gap issue too

xsharonhe commented:
> Oh yeah, I did fix the gap issue too

Didn't we resolve this last time with the padding last time? Or are you referring to a different gap issue.

xsharonhe commented on sections/hoc/PageLayout.jsx:
```diff
@@ -2,6 +2,7 @@ import { useRouter } from "next/router";
 import styled from "styled-components";
 import { Navbar } from "../Navbar";
 import { Footer } from "../Footer";
+import { media, CONSTANTS } from "../../utils";
```
Remove constants if you aren't using it

xsharonhe commented on sections/hoc/PageLayout.jsx:
```diff
@@ -22,8 +23,27 @@ export const PageLayout = ({ children, ...props }) => {
 };
 
 const Wrapper = styled.div`
+  background-image: url("/home/background-large.svg");
```
This is too specific use case to just the Top Section. There are a lot of different blobs and you should think of how to make this reusable (props) with other sections.

I also think this is not a great component to put it in. It's not really a PageLayout but a ComponentLayout. This should be a new component to wrap around each component (not each page) and you should consider the use case that the component does not have a background? If you look carefully at what PageLayout does is it wraps the children with a navbar and footer. We need image wrappers for each little section.

hirundinidae replied:
Definitely agree with Sharon, this component should be its own section wrapper (each section a page can specify which blob background svg to use).

It looks great for the top section of the home page, but even on the home page, there are other sections that require the background svg component. Every page uses the PageLayout component to include the Navbar and Footer, but not every page (or even every section) should have the same background svg as the top section of the first page 
![image](https://user-images.githubusercontent.com/65516245/128110743-ed5aabc4-9d7f-41d1-b940-885a071b6061.png)

what /resources looks like with background component in PageLayout: 
![image](https://user-images.githubusercontent.com/65516245/128110855-294cc626-c8d5-4d6f-8aef-b71f16b84ec1.png)

what /resources should look like (using different blob) 
![image](https://user-images.githubusercontent.com/65516245/128110816-76c427c9-f5a2-42ce-8ad1-881295f2eac7.png)

xsharonhe commented on sections/home/TopSection.jsx:
```diff
@@ -43,33 +43,35 @@ const MainContainer = styled.div`
         flex-direction: column-reverse;
         `
-  )};
+  )};{*/
```
Comments like this shouldn't be on here?

xsharonhe reviewed (commented):
I think you need to abstract this component more to a ComponentLayout to wrap it around each component rather than a PageLayout. I didn't exactly checkout to view on the website (since I saw it last time), but I'll leave Diana to review any of those changes for suggestions for UI.

Also next time, ask for a review on Slack on #ca-tech by pinging the reviewers (and invite other people to review too!). This makes sure we get the notification earlier.

You have to also fix merge conflicts heads up.

hirundinidae commented on sections/home/TopSection.jsx:
```diff
@@ -78,11 +80,13 @@ const JosefinTitle = styled(SText)`
     1200,
     `
         font-size: 2.5rem;
+				max-width: 100vw;
```
I'm not sure this is changing the width of the text sections. Try changing the width of the first column section @mobile breakpoint?

hirundinidae replied:
![image](https://user-images.githubusercontent.com/65516245/128112460-ef043c0c-52ba-4a62-bfbe-48bf7b7abeda.png)
vs setting wider width on tablet 
![image](https://user-images.githubusercontent.com/65516245/128112508-b57379df-4655-40b7-a85c-39ac3cb986b3.png)

hirundinidae commented on sections/home/TopSection.jsx:
```diff
@@ -43,33 +43,35 @@ const MainContainer = styled.div`
     "tablet",
     `
         line-height: 1.3;
+				max-width: 100vw;
```
Same as the title section max-width comment

hirundinidae reviewed (commented):
This works for the front page's TopSection, change to make it work for all section components. 

Try moving this component to its own SectionLayout/Wrapper component with an optional background image so each section can use something like 
```
<SectionWrapper image={src}> 
  <TopSection />  
</SectionWrapper> 
<SectionWrapper>
  <WhatsHappening /> // doesn't have a background svg 
</SectionWrapper> 
```
because the figma shows different blobs for each section of each page. 

Small UI suggestions below, there's also some widths where the text runs into the background (~1000px).

realsimonwu committed:
Create the `SectionWrapper` component

I have created the `SectionWrapper` component. As of now, to pass images to be used as backgrounds the image paths have to be relative to `/public/` - this means that the ordinary method of using `import` won't work.

It still needs some work: I may try to incorporate props that will allow for fine control of the positioning of the background image. Personally, I think it will probably be better in the long run to remake the background blobbies that already exist to accommodate already-created components, but the addition of those fine positioning controls will probably still be beneficial.

As of right now it looks TERRIBLE with pre-existing components and background blobbies. Other components will require modification (e.g. I intend to fix `TopSection` to accommodate this new background component).

realsimonwu commented:
Copied from my commit message for https://github.com/PuMProfessionals/pump-web-v2/pull/22/commits/e469cd235ba9a12fc38eae4124de8d149e324810

I have created the `SectionWrapper` component. As of now, to pass images to be used as backgrounds the image paths have to be relative to `/public/` - this means that the ordinary method of using `import` won't work.

It still needs some work: I may try to incorporate props that will allow for fine control of the positioning of the background image. Personally, I think it will probably be better in the long run to remake the background blobbies that already exist to accommodate already-created components, but the addition of those fine positioning controls will probably still be beneficial.

As of right now it looks TERRIBLE with pre-existing components and background blobbies. Other components will require modification (e.g. I intend to fix `TopSection` to accommodate this new background component).

xsharonhe commented on components/SectionWrapper.jsx:
```diff
@@ -0,0 +1,40 @@
+import React from "react";
```
Unnecessary. Next.js automatically imports React

xsharonhe commented on components/SectionWrapper.jsx:
```diff
@@ -0,0 +1,40 @@
+	background-repeat: no-repeat;
+	background-size: 100%;
+	overflow: visible;
+`;
```
Every file should have a new line ending

xsharonhe commented on components/SectionWrapper.jsx:
```diff
@@ -0,0 +1,40 @@
+  ...props
+}) => {
+	return (
+		<div>
```
This div is unnecessary

xsharonhe commented on components/SectionWrapper.jsx:
```diff
@@ -0,0 +1,40 @@
+}) => {
+	return (
+		<div>
+			<ContentWrapper
```
Add  ``` {...props} ``` here so we can change the styling if needed if it messes things up

xsharonhe reviewed (commented):
You need to fix the merge conflicts first before I can review on the home page but these are basic comments. Much better implementation of components!

realsimonwu committed:
Add in parameters for vertical and horizontal offsets

The parameters `verticalOffset`, `horizontalOffset`, `mobileVerticalOffset` and `mobileHorizontalOffset` have been added, allowing for fine positioning control of the `background-image`

realsimonwu committed:
Merge branch 'main' into add-background-graphic

realsimonwu committed:
Update index.jsx

realsimonwu committed:
Merge branch 'add-background-graphic' of https://github.com/PuMProfessionals/pump-web-v2 into add-background-graphic

realsimonwu commented:
Merge conflicts were manually resolved. https://github.com/PuMProfessionals/pump-web-v2/pull/22/commits/05fb2db0bd6249b3d3ed96469579335ce8ba280b addresses Sharon's initial recommendations and also adds the offset parameters for positioning control on the `background-image` for both mobile and full-size layouts. At present the names for these parameters are pretty long (e.g. `mobileHorizontalOffset`) so if anyone has any recommendations for shortening them then I would be open to those 😄.

Again, do note that at present the homepage looks kind of terrible with the current background blobbies provided combined with my layout, so I'll have to tweak the `TopSection` component at least. I plan on making those changes in a separate PR, though.

realsimonwu committed:
Update TopSection.jsx

xsharonhe commented on pages/index.jsx:
```diff
@@ -15,15 +20,21 @@ export default function Home() {
-        />
-        <ImpactSection />
-        <ContactSection />
+				<SectionWrapper
```
The SectionWrapper should only wrap one component and not multiple

realsimonwu replied:
A `background-image` can't extend outside of a `div`'s (or other component's) border. As a result, the current `TopSection` blobby gets cut off because of this when you only wrap `TopSection`.

Ultimately it'll probably be a good idea to modify the graphic to be shorter then it is now... for now, though, I'll just remove `SectionWrapper` from the homepage until we sort something out.

xsharonhe replied:
Yep, I realized that so I think we need to contact design to change the blob. Rachel Chen is the one who made this page so it would be best to contact her. And sounds good.

xsharonhe reviewed (commented):
I think theoretically this is okay, but it should only wrap around one component, or at minimum two components because we need to wrap other sections. It looks like we need to remake the svg in general because the overlapping issue with text seems pretty big. Is there anyway to get background height to be 100% (making the TopSection height bigger in general... for desktop). 

Can you contact Rachel Chen on #ca-tech about changing this graphic? It works fine on larger desktop width and mobile width but should overlap less with the text.

We can merge this SectionWrapper but I think for now let's not have the implementation on TopSecetion because there's still some slight problems? Idk if Diana has more thoughts so we can wait, but the overall SectionWrapper component looks good

realsimonwu committed:
Update index.jsx

xsharonhe reviewed (dismissed):
LGTM. Remember to squash commits :D

realsimonwu committed:
Prettier fixes

realsimonwu committed:
Merge branch 'add-background-graphic' of https://github.com/PuMProfessionals/pump-web-v2 into add-background-graphic

realsimonwu commented:
I pushed another commit to make Prettier happy, could you re-approve? 😅

xsharonhe reviewed (approved):
Don't worry about prettier I can fix on my ticket :)
